fuel/transformers/__init__.py

```diff
@@ -97,8 +114,9 @@ class Mapping(Transformer):
         data under source names `add_sources`.
 
     """
-    def __init__(self, data_stream, mapping, add_sources=None):
-        super(Mapping, self).__init__(data_stream)
+    def __init__(self, data_stream, produces_examples, mapping,
```

Comment:
Actually, we don't even need to do that (and shouldn't) because the
overrided property's non-existent setter will get called. So that will
catch it if someone tries to pass `produces_examples`.

On Thu, Jul 9, 2015 at 1:33 PM, vdumoulin <notifications@github.com> wrote:

> In fuel/transformers/__init__.py
> <https://github.com/mila-udem/fuel/pull/188#discussion_r34287660>:
>
> > @@ -97,8 +114,9 @@ class Mapping(Transformer):
> >          data under source names `add_sources`.
> >
> >      """
> > -    def __init__(self, data_stream, mapping, add_sources=None):
> > -        super(Mapping, self).__init__(data_stream)
> > +    def __init__(self, data_stream, produces_examples, mapping,
>
> Although we could simply call the superclass' constructor by passing
> self.data_stream.produces_examples. An error will be raised if someone
> tries to pass produces_examples as a kwarg.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/fuel/pull/188/files#r34287660>.
>